neqo-transport/src/cc.rs

```diff
@@ -103,10 +99,16 @@ impl CongestionControl {
             assert!(self.bytes_in_flight >= pkt.size);
             self.bytes_in_flight -= pkt.size;
 
-            if self.in_congestion_recovery(pkt.time_sent, PacketState::Acked) {
+            if self.in_congestion_recovery(pkt.time_sent) {
                 // Do not increase congestion window in recovery period.
                 continue;
             }
+
+            if self.in_recovery {
+                self.in_recovery = false;
+                qlog::metrics_updated(&mut self.qlog, &[QlogMetric::InRecovery(false)]);
+            }
```

Comment:
Can't we just set the value of congestion_recovery_start_time to `None` at this point?

Reply:
I think we need to keep the value to be able to detect that a loss is in the old recovery phase even in case of reordering.

Reply:
Yeah this was the bug. We lost a packet, and therefore set CRST to now (T=1). Declaring another packet lost that was sent before CRST should not affect CC even if we have exited recovery (gotten an ack for a packet sent after CRST.)  Receiving an ack at T=2 for a packet sent after T=1 will cause CC to exit recovery. It's still possible for that or another ack received at T=3 to cause a sent packet(A), which previously didn't meet the packet reorder threshold (3), to be declared lost. If we forget previous CRST then declaring that packet lost will set CRST to T=3 and start a new recovery period, halves cwnd, etc. If CRST still is T=1 then we can tell packet A was sent before T=1.

The fact that CRST is an `Option<Instant>` is really only because there's no valid `Instant` value we can put there before the first time we lose a packet, causing recovery to happen.